lib/subscription/subscription-store.ts

```diff
@@ -0,0 +1,72 @@
+/// <reference path='../../typings/tsd.d.ts' />
+
+import _ = require('lodash');
+import Subscription = require('./subscription');
+
+export = SubscriptionStore;
+
+class SubscriptionStore<T extends Subscription> {
```

Comment:
This class encapsulate some common data retrieval/look up functions and I would suggest to keep it. It will make more sense for the long-polling version where I extend from this class and add buffering related operations into it. The generic is needed so the child class can supply the type of `Subscription` they operate on. This is what I have for the long-polling:

```typescript
class Subscription {}
class SubscriptionWithBuffer extends Subscription{}

class SubscriptionStore<T extends Subscription> {}
class SubscriptionStoreWithBuffer extends SubscriptionStore<SubscriptionWithBuffer>
```